lib/Doctrine/ORM/Mapping/ClassMetadata.php

```diff
@@ -438,7 +438,7 @@ class ClassMetadata implements PersistenceClassMetadata, Stringable
      * )
      * </pre>
      *
-     * @psalm-var array<string, OneToOneAssociationMapping|ManyToOneAssociationMapping|OneToManyAssociationMapping|ManyToManyAssociationMapping>
+     * @psalm-var array<string, AssociationMapping>
```

Comment:
I had to revert to something more simple here. It confused Psalm too much, especially when performing intersections with owning types.